src/java/org/apache/nutch/plugin/URLStreamHandlerFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.apache.nutch.plugin;
+
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.net.URLStreamHandler;
+import java.util.ArrayList;
+
+import org.mortbay.log.Log;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class URLStreamHandlerFactory
+    implements java.net.URLStreamHandlerFactory {
+  
+  protected static final Logger LOG = LoggerFactory
+      .getLogger(URLStreamHandlerFactory.class);
+  
+  /** The singleton instance. */
+  private static URLStreamHandlerFactory instance;
+  
+  /** Here we register all PluginRepositories. */
+  private ArrayList<WeakReference<PluginRepository>> prs;
```

Comment:
This is interesting, and i think it may be the first example of WeakReference usage within the Nutch source. Can you briefly explain your justification for it I am keen to know.